test/linter.js

```diff
@@ -4,23 +4,126 @@
+    // get path to the modified files relative to their corresponding md file, need to do this since 
+    // config files have relative paths to the input files
+    let allModifiedFiles = utils.getFilesChangedInPR();
+    allModifiedFiles = allModifiedFiles.map(mfile => {
+      return mfile.replace(path.dirname(config) + '/', '');
+    });
+
+    // for each tag->files, find if there are any modified files and select those tags
+    return tags.filter(tag => {
+      
+      const tagFiles = (String(tagsMap[tag])).split(',');
+      // find intersection with the modified files
+      return tagFiles.filter(tagFile => {
+        return allModifiedFiles.indexOf(tagFile) > -1;
+      }).length > 0;
+    });
+  }
+  return tags;
+}
+
+function execLinterCommand(config, tag) {
+  var cmd = `autorest --validation --azure-validator ${config} --message-format=json ${tag}`.trim();
+  console.log(`Executing: ${cmd}`);
+  try {
+    let result = execSync(cmd, { encoding: 'utf8', maxBuffer: 1024 * 1024 * 64 });
+  } catch (err) {
+    throw new Error('Linter validation contains error(s)');
+  }
+
+}
 
 describe('AutoRest Linter validation:', function () {
-  let configsToProcess = utils.getConfigFilesChangedInPR();
   // Useful when debugging a test for a particular swagger. 
   // Just update the regex. That will return an array of filtered items.
   // configsToProcess = ['specification/sql/resource-manager/readme.md'];
+  let configsToProcess = utils.getConfigFilesChangedInPR();
   for (const config of configsToProcess) {
     it(config + ' should honor linter validation rules.', async function () {
-      var cmd = `autorest --validation --azure-validator ${config} --message-format=json`;
-      console.log(`Executing: ${cmd}`);
 
-      try {
-        let result = execSync(cmd, { encoding: 'utf8', maxBuffer: 1024 * 1024 * 64 });
-      } catch (err) {
-        throw new Error('Linter validation contains error(s)');
+      // find all tags in the config file
+      const tagsToProcess = await getTagsFromConfig(config);
+      // if no tags found to process, run with the defaults    
+      if (tagsToProcess === null || tagsToProcess.length === 0) {
+        // no tags found
+        // this means we need to run validator against the individual 
+        // json files included in the PR
+        // but in the same directory tree as the config file
+        const filesChangedInPR = utils.getFilesChangedInPR();
+        const configDir = path.dirname(config);
+        filesChangedInPR.filter(prfile => {
+          // set any type to string
+          prFile += '';
+          return prFile.startsWith(configDir) && prfile.indexOf('examples') === -1 && prFile.endsWith('.json');
+        }).forEach(prFileInConfigFile => {
+          console.warn(`Configuration file not found for file: ${prFileInConfigFile}, running validation rules against it in individual context.`);
+          execLinterCommand(`--input-file=${prFileInConfigFile}`, '');
```

Comment:
Since `execLinterCommand` does nothing smart about its two parameters (they're just interpolated into the command to run... you could pass `"--csharp"` as `tags`), I think the signature should be `execLinterCommand(...args: string[])`, then you can call it as 
```
execLinterCommand(`--input-file=${prFileInConfigFile}`)
```

More readable and there's no parameter names that suggest semantics that don't exist.